### PR TITLE
feat(uiSelectSingleDirective): add an option to avoid backspace model reset

### LINKED ISSUE #926: Prevent backspace key from clearing selected value
Is there a way to prevent backspace key from clearing the selected value in ui-select?

Thank you in advance

### PR DESCRIPTION
**Pull request by pierregaste**
Add a new option that prevent the backspace key reset the model (set to undefined).
Th option could be added to the directive `ui-select`:

``` html
<ui-select disable-backspace-reset="true"></ui-select>
```

I hope I correctly commit and implement this feature! :smiley: 
---

feat(uiSelectSingleDirective): add an option to avoid backspace model reset

Currently, when you select an option and you press the backspace key, it will reset the model.
With this option, we could disable this behavior to avoid the model resetting.

Closes #926 #525
---

This line could be added to the wiki page (uiSelect):

```
| `backspace-reset` | Set to false to avoid resetting the model when the backspace is pressed  | boolean | true |
```

### CODE REVIEW

**Comment by pierregaste** [comment]
I removed my previous fork by mistake.
https://github.com/angular-ui/ui-select/pull/1481/commits/d94f07631067963db171bcda07ddb01cd9299914

File: src/uiSelectDirective.js [hunk 1]
```diff
@@ -83,6 +83,11 @@ uis.directive('uiSelect',
             $select.sortable = sortable !== undefined ? sortable : uiSelectConfig.sortable;
         });
 
+        scope.$watch('disableBackspaceReset', function() {
```
Comment (user378230) [inline src/uiSelectDirective.js]: Shouldn't you be watching the attribute?

**Comment by LeleDev** [comment]
eagerly waiting for this to be merged

File: src/uiSelectDirective.js (hunk 1 above)
Comment (pierregaste) [inline src/uiSelectDirective.js]: Yes, sorry. I fixed it.

**Comment by chardea7** [comment]
we need this compononen.. please merge it ASAP thanks

**Comment by pierregaste** [comment]
Any news about my PR? I just sync the fork.

Thanks for your answer! 😃

**Comment by user378230** [comment]
@pierregaste Instead of calling it `disableBackspaceReset` can we call it `backspaceReset` set default to `true`, then user can specify attribute `backspace-reset="false"` to stop?

If you do this and update tests etc. I promise we will merge! You appear to have unrelated changes in your commits can you resolve this. Also would help if you squash to single commit for us. Thanks for your patience!

File: src/uiSelectDirective.js [hunk 2]
```diff
@@ -83,8 +83,8 @@ uis.directive('uiSelect',
             $select.sortable = sortable !== undefined ? sortable : uiSelectConfig.sortable;
         });
 
-        scope.$watch(function () { return scope.$eval(attrs.disableBackspaceReset); }, function(newVal) {
-          $select.disableBackspaceReset = newVal !== undefined ? newVal : uiSelectConfig.disableBackspaceReset;
+        scope.$watch(function () { return scope.$eval(attrs.backspaceReset); }, function(newVal) {
```
Comment (user378230) [inline src/uiSelectDirective.js]: Can you use `attrs.$observe(...)` here instead? Only `$eval` when value is different. (See `resetSearchInput` a few lines down)

**Comment by user378230** [comment]
@pierregaste I just added one more comment too, thanks!

File: src/uiSelectDirective.js [hunk 3]
```diff
@@ -83,6 +83,10 @@ uis.directive('uiSelect',
             $select.sortable = sortable !== undefined ? sortable : uiSelectConfig.sortable;
         });
 
+        scope.$watch(function () { return scope.$eval(attrs.backspaceReset); }, function(newVal) {
```
Comment (user378230) [inline src/uiSelectDirective.js]: Can you use attrs.$observe(...) here instead? Only $eval when value is different. (See resetSearchInput a few lines down)

**Comment by pierregaste** [comment]
@user378230 I replace the $watch method by the $observe method. But I don't know how to squash and rebase. Could you help me to do this?

**Pull request merged**